src/System.Diagnostics.DiagnosticSource/src/HierarchicalRequestId.md

```diff
@@ -0,0 +1,110 @@
+#  Hierarchical Request-Id
+This document describes hierarchical Request-Id schema for [HTTP Correlation Protocol](HttpCorrelationProtocol.md) for telemetry correlation.
+
+## Overview
+The main requirement for Request-Id is uniqueness, any two requests processed by the cluster must not collide.
+Guids or big random number help to achieve it, but they require other identifiers to query all requests related to the operation.
+
+Hierarchical Request-Id look like `|<root-id>.<local-id1>.<local-id2>.` (e.g. `|9e74f0e5-efc4-41b5-86d1-3524a43bd891.bcec871c_1.`) and holds all information needed to trace whole operation and particular request.
+Root-id serves as common identifier for all requests involved in operation processing and local-ids represent internal activities (and requests) done within scope of this operation.
+
+[CorrelationVector](https://osgwiki.com/wiki/CorrelationVector) is valid hierarchical Request-Id, except it does not start with "|". Implementation SHOULD allow other schemes for incoming request identifiers.
```

Comment:
I don't think this link is available from outside of MS. Please remove or copy the content publicly. 

CC: @jacpull